src/test/uint256_tests.cpp

```diff
@@ -386,6 +386,33 @@ BOOST_AUTO_TEST_CASE(from_hex)
     TestFromHex<Wtxid>();
 }
 
+BOOST_AUTO_TEST_CASE(from_user_hex)
+{
+    constexpr std::string_view valid_hex_65{"0x0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef0"}; // 0x prefix and 65 hex chars
+    static_assert(valid_hex_65.size() == 2 + 65);
+
+    BOOST_CHECK_EQUAL(uint256::FromUserHex("").value(), uint256::ZERO);
+    BOOST_CHECK_EQUAL(uint256::FromUserHex("0x").value(), uint256::ZERO);
+    BOOST_CHECK_EQUAL(uint256::FromUserHex("0").value(), uint256::ZERO);
```

Comment:
nit: I was wondering if it accepts `00` as well (since numbers usually trim leading zeros) and if it accepts `1` (since other implementations require an even number of digits), consider extending the examples:
```C++
    BOOST_CHECK_EQUAL(uint256::FromUserHex("00").value(), uint256::ZERO);
    BOOST_CHECK_EQUAL(uint256::FromUserHex("1").value(), uint256::ONE);
```

Reply:
Thanks, I've added these two test cases!